jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -98,38 +95,21 @@ def process(self, input, *args, **kwargs):
         if not (result.meta.exposure.type.lower() in IMAGING_TYPES and self.sip_approx):
             return result
 
+        (result, )
+
         # fit sip approx., degree is chosen by best fit
-        sip_degree = range(1, _MAX_SIP_DEGREE) if self.sip_degree is None else self.sip_degree
-        sip_inv_degree = range(1, _MAX_SIP_DEGREE) if self.sip_inv_degree is None else self.sip_inv_degree
-        crpix = [result.meta.wcsinfo.crpix1, result.meta.wcsinfo.crpix2]
-        crpix = None if None in crpix else crpix
         try:
-            fit_sip_hdr = result.meta.wcs.to_fits_sip(
+            update_fits_wcsinfo(
+                result,
                 max_pix_error=self.sip_max_pix_error,
-                degree=sip_degree,
+                degree=self.sip_degree,
                 max_inv_pix_error=self.sip_max_inv_pix_error,
-                inv_degree=sip_inv_degree,
+                inv_degree=self.sip_inv_degree,
                 npoints=self.sip_npoints,
-                crpix=crpix
+                crpix=None
             )
 
         except ValueError:
```

Comment:
should there be an error message if the wcsinfo is unable to be updated?

Reply:
I don't know.. I kept the behavior exactly as it was in the code that you wrote. No changes. Why did you wrap `to_fits_sip()` in a `try-except` block? I simply do not know the history behind this decision. @cshanahan1 @nden

Reply:
was this something i did? if so i dont remember why haha. let me think about this for a bit and ill amend my comment if needed.

Reply:
Well, I added a log warning message if assigning `wcsinfo` fails (in a new commit). I think on failure to do what is expected it would be nice to get a warning.